keps/sig-node/3545-improved-multi-numa-alignment/README.md

```diff
@@ -122,7 +125,7 @@ This limitation surfaces in multi socket, as well as single socket multi NUMA sy
 ### Proposed Change
  
 We propose to
-- add a new flag in Kubelet called `TopologyManagerPolicyOptions` in the kubelet config or command line argument called `topology-manager-policy-options` which allows the user to specify the Topology Manager policy option.
+- add a new flag in Kubelet called `TopologyManagerPolicyAlphaOptions` in the kubelet config or command line argument called `topology-manager-policy-options` which allows the user to specify the Topology Manager policy option.
```

Comment:
I think the flag should still be called `TopologyManagerPolicyOptions` (corresponding to `topology-manager-policy-options` flag in kubelet). 

May be we are confusing flags with feature gates? Kubelet flags are different from Feature gates (the same name here is probably not helping!). In this case we we would need a kubelet flag (`TopologyManagerPolicyOptions` ) and three feature gates:

1.  `TopologyManagerPolicyOptions` for enabling/disabling the entire feature. As this is a brand new feature introduced as alpha, the feature gate would be disabled by default. Explicitly enabling `TopologyManagerPolicyOptions` feature gate provides us the ability to supply `topology-manager-policy-options` flag in Kubelet. 
2.  `TopologyManagerPolicyAlphaOptions` to enable/disable the visibility of options at alpha level options. These options would be hidden by default.
3.  `TopologyManagerPolicyBetaOptions` to enable/disable the visibility beta level. These options would be visible by default.

IOW, to use `topology-manager-policy-options=prefer-closest-numa-nodes`  in Kubernetes 1.26 we would have to specify both `TopologyManagerPolicyOptions` and `TopologyManagerPolicyAlphaOptions`. Once we graduate the overall feature to Beta `TopologyManagerPolicyOptions` is enabled by default and `prefer-closest-numa-nodes` moves from an alpha option to a Beta option (controlled by feature gate`TopologyManagerPolicyBetaOptions` feature gate) and is therefore visible by default without the need to providing a feature gate. In case of Beta options, `TopologyManagerPolicyBetaOptions` can be used to disable these policy options.

Hope this helps and provides more clarity.

In my comment [here](https://github.com/kubernetes/enhancements/pull/3549#discussion_r985636985), I was suggesting that we could get away with a single feature gate as this is an alpha feature and we only have a single option but it probably makes sense to capture the feature gates for the options to indicate maturity level of options right from the beginning to document and set expectations for options proposed in the future.

Reply:
I agree with everything stated here. It's not stated explicitly here, but I want to emphasize that we should actually add the feature gate for `TopologyManagerPolicyBetaOptions` **now** (even though there is nothing to gate this with option yet). That way there is less code churn once we do graduate this feature to beta in a future release.

Reply:
Fixed.